3.0/m_roleplay.cpp

```diff
@@ -427,25 +477,32 @@ class CommandBaseRoleplay : public Command
 		MessageDetailsImpl msgdetails(MSG_PRIVMSG, GetMessage(parameters), parameters.GetTags());
 		MessageTarget msgtarget(c, 0);
 
+		source = MakeFakeHostmask(user, source);
+
+		/* Put the tags on as soon as possible just in case anything
+		 * wants to look at it.
+		 */
+		msgdetails.AddTag("inspircd.org/roleplay-msg", &roleplaymsgtag, user->GetFullHost());
+		msgdetails.AddTag("inspircd.org/roleplay-src", &roleplaysrctag, source);
+
 		if(!CheckMessage(user, msgtarget, msgdetails))
 			return CMD_FAILURE;
 
-		// Spoof the hostmask for the source nickname before sending
-		source = MakeFakeHostmask(user, source);
+		// Do the thing
 		SendMessage(user, c, source, msgtarget, msgdetails);
 
-		/* Since this is a message, if the user is local, then update
-		 * their idle time.
-		 */
-		if(luser)
-			luser->idle_lastmsg = ServerInstance->Time();
+		// Since this is a message, update the users' idle time.
+		user->idle_lastmsg = ServerInstance->Time();
 
 		return CMD_SUCCESS;
 	}
 
 	RouteDescriptor GetRouting(User* user, const CommandBase::Params& parameters) CXX11_OVERRIDE
 	{
-		return ROUTE_OPT_BCAST;
+		/* spanningtree will broadcast this as a PRIVMSG
+		 * It would be actively harmful to broadcast this.
+		 */
+		return ROUTE_LOCALONLY;
```

Comment:
ROUTE_LOCALONLY Is the default so you can omit GetRouting.

Reply:
Done.